src/lib/dnssd/Resolver.h

```diff
@@ -108,53 +156,15 @@ struct DiscoveredNodeData
     size_t rotatingIdLen;
     uint16_t pairingHint;
     char pairingInstruction[kMaxPairingInstructionLen + 1];
-    bool supportsTcp;
-    Optional<System::Clock::Milliseconds32> mrpRetryIntervalIdle;
-    Optional<System::Clock::Milliseconds32> mrpRetryIntervalActive;
-    uint16_t port;
-    unsigned numIPs;
-    Inet::InterfaceId interfaceId; // interface ID for the packet processed
-    Inet::IPAddress ipAddress[kMaxIPAddresses];
 
     void Reset()
     {
-        memset(hostName, 0, sizeof(hostName));
-        memset(instanceName, 0, sizeof(instanceName));
-        longDiscriminator = 0;
-        vendorId          = 0;
-        productId         = 0;
-        commissioningMode = 0;
-        deviceType        = 0;
-        memset(deviceName, 0, sizeof(deviceName));
-        memset(rotatingId, 0, sizeof(rotatingId));
-        rotatingIdLen = 0;
-        memset(pairingInstruction, 0, sizeof(pairingInstruction));
-        pairingHint            = 0;
-        supportsTcp            = false;
-        mrpRetryIntervalIdle   = NullOptional;
-        mrpRetryIntervalActive = NullOptional;
-        numIPs                 = 0;
-        port                   = 0;
-        for (unsigned i = 0; i < kMaxIPAddresses; ++i)
-        {
-            ipAddress[i] = chip::Inet::IPAddress::Any;
-        }
+        // NOTE that this assumes 0-fill is ok for this class
+        memset(this, 0, sizeof(*this));
```

Comment:
Would it be better if the constructor was conventionally initializing members, and then Reset() called the destructor + placement new?

Reply:
Was unsure - placement new/destructor feels a bit hackish.

Is there a benefit? I was assuming memset 0 is very fast and this is what the previous code was doing anyway.

Reply:
I think my critique starts with the fact that the constructor was indirectly calling memset, and that there had to be a comment in the code explaining why this is safe under limited conditions.  It just seems that if this is what we mean to do, we should use c++ tools to do it.  This way, if the class becomes non-trivial in the future, things won't break.